packages/manager/apps/dedicated/client/app/dedicated/dedicated-server/clusters/cluster/cluster.service.js

```diff
@@ -22,4 +26,23 @@ export default class Cluster {
       .get(`/services/${serviceId}`)
       .then(({ data }) => data?.billing?.plan?.invoiceName);
   }
+
+  /**
+   * Retrieve all nodes from a cluster
+   * @param {String} serviceName of the cluster (cluster ID)
+   * @returns {DedicatedServer[]} list of nodes of the cluster
+   */
+  async getNodes(serviceName) {
+    const cluster = await this.getSelected(serviceName);
+    const serverNames = cluster.nodes?.map(({ serverName }) => serverName);
+    const nodes = await this.$q.all(
+      serverNames.map((serverName) =>
+        this.Server.getSelected(serverName).then(
+          (swsResponse) => new DedicatedServer(swsResponse),
+        ),
+      ),
+    );
+
+    return nodes;
```

Comment:
**nitpick:** Is `async` &. `await` working as expected because I have noticed an issue few times where we had to explicitly trigger the Digest cycle to update the view.
IMO, better to switch to `.then`

Reply:
In this case it should work as expected because it is used inside resolve so digest cycle is triggered anyway at component init